### PR TITLE
Client server test coverage

### PR DESCRIPTION
**Pull request by Mathixx**
Added profile CRUD test for MatrixDb to ensure correct test coverage

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,4 @@ packages/*/example/*.js*
 *.map
 .nx/
 packages/matrix-identity-server/matrix-server/synapse-data
+.vscode
```
Comment (chibenwa) [inline .gitignore:8]: I'm not forced to use VS code and we likely do not want to add and maintain one line per personnal preference.

Generally editor settings goes onto personal global gitignores not onto the project ones that generally only excludes build materials.

CF https://gist.github.com/subfuzion/db7f57fff2fb6998a16c

**Review by chibenwa — COMMENTED** [review]
Good to me. FYI I am not a JS expert and Matrix is an unknown topic to me.

Maybe be more precise onto the commit / PR message: we are only testing profile CRUD here, not the whole matrix DB.

File: .gitignore (hunk 1 above)
Comment (Mathixx) [reply .gitignore:8]: You're write I'll fix that
Thanks !

File: packages/matrix-client-server/src/matrixDb/matrixDbTestConf.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  "rate_limiting_window": 10000,
+  "server_name": "matrix.org",
+  "smtp_sender": "yadd@debian.org",
+  "smtp_server": "localhost",
+  "template_dir": "./templates",
+  "userdb_engine": "sqlite",
+  "userdb_host": "./src/__testData__/test.db",
+  "matrix_database_engine": "sqlite",
+  "matrix_database_host": "./src/__testData__/testMatrix.db"
+}
```
Comment (guimard) [inline packages/matrix-client-server/src/matrixDb/matrixDbTestConf.json:20]: Such file must be in a __something__ directory _(test only)_

Comment (Mathixx) [reply packages/matrix-client-server/src/matrixDb/matrixDbTestConf.json]: Moved it to the __testData__ folder !

**Comment by guimard** [comment]
Test still broken with insufficient coverage :
```
       -----------------------------------|---------|----------|---------|---------|-------------------
       File                               | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
       -----------------------------------|---------|----------|---------|---------|-------------------
       All files                          |   73.02 |    51.35 |   55.55 |   72.98 |                   
        src                               |   94.11 |      100 |   72.72 |      94 |                   
         index.ts                         |    92.1 |      100 |   66.66 |   91.89 | 43-50             
         types.ts                         |     100 |      100 |     100 |     100 |                   
        src/account                       |     100 |       50 |      80 |     100 |                   
         whoami.ts                        |     100 |       50 |      80 |     100 | 16                
        src/admin                         |     100 |      100 |   88.88 |     100 |                   
         whois.ts                         |     100 |      100 |   88.88 |     100 |                   
        src/matrixDb                      |     100 |      100 |     100 |     100 |                   
         index.ts                         |     100 |      100 |     100 |     100 |                   
        src/matrixDb/sql                  |   92.59 |    33.33 |   91.66 |   95.83 |                   
         sqlite.ts                        |   92.59 |    33.33 |   91.66 |   95.83 | 87                
        src/utils                         |   37.64 |    34.78 |   14.28 |   37.64 |                   
         authenticate.ts                  |     100 |      100 |     100 |     100 |                   
         userInteractiveAuthentication.ts |   11.66 |        0 |    3.22 |   11.66 | 30-118,130-209    
       -----------------------------------|---------|----------|---------|---------|-------------------
       Jest: "global" coverage threshold for statements (90%) not met: 73.02%
       Jest: "global" coverage threshold for branches (80%) not met: 51.35%
       Jest: "global" coverage threshold for lines (90%) not met: 72.98%
       
       Test Suites: 2 passed, 2 total
       Tests:       15 passed, 15 total
       Snapshots:   0 total
       Time:        7.463 s, estimated 9 s
```

File: .gitignore (hunk 1 above)
Comment (chibenwa) [reply .gitignore]: > You're write

You're right ;-)

**Pull request merged**